runbotics/runbotics-orchestrator-ui/src/src-app/components/CredentialSelectSet/CredentialSelectSet.tsx

```diff
@@ -0,0 +1,100 @@
+import { ChangeEvent, FunctionComponent, useEffect, useMemo, useState } from 'react';
+
+import { MenuItem, Typography } from '@mui/material';
+import _ from 'lodash';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+import { Credential } from '#src-app/store/slices/Credentials';
+
+import { Box, StyledSelect } from './CredentialSelectSet.styles';
+
+interface CredentialSelectSetProps {
+    credentials: Credential[];
+    handleCredentialChange: (credential: Credential | undefined) => void;
```

Comment:
If find function in this component will not find anything it will return undefined and then this handle will contain undefined value. But I will simplify a bit